### ISSUE TITLE
Allow 'object of objects' for 'ElectricalStimulationParameters'

### ISSUE THREAD

**Issue by richardkoehler** [bids-standard/bids-specification#994]
I would like to propose to allow the data type `object of objects` additionally to `string` for `ElectricalStimulationParameters` in the `*ieeg.json` file ([see current specification](https://bids-specification.readthedocs.io/en/latest/04-modality-specific-files/04-intracranial-electroencephalography.html#sidecar-json-_ieegjson)).

To give you an impression of what this would allow for, this is an example of the standard that our collaborators and us use to share electrical stimulation parameters:

```
"ElectricalStimulationParameters": {
        "BestClinicalSetting": "n/a",
        "CurrentExperimentalSetting": {
	        "DateOfSetting": "2021-11-12",
	        "StimulationTarget": "STN",
	        "StimulationMode": "continuous",
	        "StimulationParadigm": "continuous stimulation",
	        "SimulationMontage": "monopolar",
	        "Left": {
		        "AnodalContact": "G",
		        "CathodalContact": "7, 8 and 9",
		        "AnodalContactDirection": "none",
		        "CathodalContactDirection": "omni",
		        "CathodalContactImpedance": "n/a",
		        "StimulationAmplitude": 2,
		        "StimulationPulseWidth": 60,
		        "StimulationFrequency": 130,
		        "InitialPulseShape": "rectangular",
		        "InitialPulseWidth": 60,
		        "InitialPulseAmplitude": -2,
		        "InterPulseDelay": 0,
		        "SecondPulseShape": "rectangular",
		        "SecondPulseWidth": 60,
		        "SecondPulseAmplitude": 2,
		        "PostPulseInterval": "n/a"
	        }
        }
}
```

I would propose something similar to this for the BIDS Specification:

| **Key name** | **Data type** | **Description**                                          |
|--------------|---------------|----------------------------------------------------------|
| ElectricalStimulationParameters | object of objects or string | Description of stimulation parameters, such as frequency or shape. Can be key:value pairs (for example, `{"StimulationAmplitude": 2, "StimulationPulseWidth": 60, "StimulationFrequency": 130}` or free form (for example, `"Stimulation amplitude was 2.0 mA, pulse width 60 µs and stimulation frequency was 130 Hz."`. Stimulation onsets and durations can be specified in the events.tsv file.|

**Comment by JonathanVHoecke** [comment]
I endorse this issue of my colleague @richardkoehler, as we brainstormed about this across different neuroscience departments in Germany.

**Comment by sappelhoff** [comment]
Thanks for the proposal, and good to hear that you have already discussed this across many departments. Tagging @bids-standard/raw-electrophys-ieeg to give this proposal a look/review.

**Comment by JonathanVHoecke** [comment]
Thank you very much Stefan, we  really appreciate that!

**Comment by adam2392** [comment]
Is there any chance to be more descriptive and explicit instead of just allowing an `object of objects`? I suppose not if stimulation isn't really standardized, or could this be an opportunity to standardize?

Besides that, extending the ieeg stimulation section with an example such as the above that is described would be really helpful as well!

LGTM.

**Comment by JonathanVHoecke** [comment]
Thank you for your enthusiasm.

In further consideration about this idea, we unfortunately need to channel this back to the consortium first.
The concept of having an object of object would be really great, but the precise description you refer to as written by @richardkoehler is actually subject of further internal discussion with our collaborators.

We are happy to remain in touch about this!

**Comment by richardkoehler** [comment]
@adam2392 
I fully agree that it would be best to be even more explicit. 
However, stimulation settings are close to infinite, and it will be a pretty large task to cover all possible cases.
Electrical stimulation in BIDS would probably even require a separate BEP.
This is something we could definitely consider and something I would be happy to work on.
In the meantime, adding the `object of objects` would allow to share all the stimulation metadata that one might wish to share.

**Comment by dorahermes** [comment]
While electrical stimulation is a tricky issue, as there are many parameters (mono/bipolar, mono/biphasic, pulse shape...), the parameter space seems graspable to me. Having stimulation amplitude 2 does not really mean anything if you don't have units.

@robertoostenveld What is the current downstream effect if an object of objects would be allowed in the current version, but a future BEP would restrict the fields?

**Comment by robertoostenveld** [comment]
Many of the stimulation techniques are not exclusive for a specific modality for recording brain activity. @GiacomoBertazzoli and @viacovella have recently started a draft for a new BIDS extension proposal to better document (non-invasive) brain stimulation. I think it would be relevant to consider their ideas on this. Not only intracranial EEG stimulation, but also stimulation using scalp electrodes (aka TACS) or magnetic field coils (aka TMS) is complex to describe.

**Comment by GiacomoBert** [comment]
Yes, indeed we are working on a draft BEP called NIBS-BIDS (non-invasive brain stimulation BIDS) since last summer. It is almost ready to be shared with the community. Like you, we are also facing the issue of describing a setting that has such a big parameter choice and combination with other techniques. We thought about possible solutions and it would be great to discuss it with you. I will update you as soon as we share it.

**Comment by richardkoehler** [comment]
@GiacomoBert 
That is really cool, I am very much looking forward to reading your proposal and discussing potential solutions!
It would be really cool if the standards for non-invasive and invasive stimulation were as similar as it is possible.
